setup.py

```diff
@@ -35,6 +38,44 @@ def run(self):
         os.system('rm -vrf ./build ./dist ./*.pyc ./*.tgz ./*.egg-info')
 
 
+def git_version(version):
+    try:
+        import git
+    except ImportError:
+        logger.warn('gitpython not found: Cannot compute the git version.')
+        return ''
+    try:
+        repo = git.Repo('.git')
+    except ImportError:
+        logger.warn('Git repo not found: Cannot compute the git version.')
+        return ''
+    sha = repo.head.commit.hexsha
+    if repo.is_dirty():
+        return '.dev0+{sha}.dirty'.format(sha=sha)
+    # commit is clean
+    # is it release of `version` ?
+    try:
+        tag = repo.git.describe(
+            match='[0-9]*', exact_match=True,
+            tags=True, dirty=True)
+        assert tag == version, (tag, version)
+        return '.release:{version}+{sha}'.format(version=version,
+                                                 sha=sha)
+    except git.GitCommandError:
+        return '.dev0+{sha}'.format(sha=sha)
+
+
+def write_version(filename=os.path.join('airflow', 'git_version')):
+    cnt = """%(git_revision)s"""
+    text = cnt % {'git_revision':
```

Comment:
the python community is moving away from the modulo format operator in favor of `"{}".format()` function.